provider/plugin/plugin.go

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"sigs.k8s.io/external-dns/endpoint"
+	"sigs.k8s.io/external-dns/plan"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type PluginProvider struct {
+	client          *http.Client
+	remoteServerURL *url.URL
+}
+
+type PropertyValuesEqualsRequest struct {
+	Name     string `json:"name"`
+	Previous string `json:"previous"`
+	Current  string `json:"current"`
+}
+
+type PropertiesValuesEqualsResponse struct {
+	Equals bool `json:"equals"`
+}
+
+func NewPluginProvider(u string) (*PluginProvider, error) {
+	parsedURL, err := url.Parse(u)
+	if err != nil {
+		return nil, err
+	}
+	return &PluginProvider{
+		client:          &http.Client{},
+		remoteServerURL: parsedURL,
+	}, nil
+}
+
+// Records will make a GET call to remoteServerURL/records and return the results
+func (p PluginProvider) Records(ctx context.Context) ([]*endpoint.Endpoint, error) {
```

Comment:
As a possible future optimization the GET call include a hash of the current records as a parameter, so the plugin provider service doesn't need to send the records back if nothing has changed.

Reply:
I'll think about it, but this is a call on localhost and the data transferred is relatively little, I don't think there is a lot to optimize. We're always talking about DNS which is something with relatively long delays by nature, so optimizations aren't my first thought.